src/server/authMiddleware.ts

```diff
@@ -72,6 +93,13 @@ const createAuthMiddleware =
 	async (req: NextRequest) => {
 		console.debug('Auth middleware starts');
 
+		if (options.publicRoutes && options.privateRoutes) {
```

Comment:
what if private routes are empty array? they will return true as well

we should be more permissive  

```suggestion
		if (options.publicRoutes?.length > 0 && options.privateRoutes?.length > 0) {
```